### PR TITLE
dependency: update mocha-junit-reporter to 2.2.0

### PR DESCRIPTION
**Pull request by AlexGuironnetRTE**
- Closes 

### Additional details
[Version 2.2.0](https://github.com/michaelleeallen/mocha-junit-reporter/releases/tag/v2.2.0) of mocha-junit-reporter provides [new placeholders](https://github.com/michaelleeallen/mocha-junit-reporter/pull/163) to define the report file name

### Steps to test

N/A

### How has the user experience changed?

N/A

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
@AlexGuironnetRTE Changes look good! Can you add a changelog entry for the dependency update so users are away this has bumped?

**Comment by AlexGuironnetRTE** [comment]
Thanks for the review! I added the changelog entry, sorry I missed it. I didn't realize that a dependency update qualified as a user-facing change but that makes sense.

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+
+## 12.7.1
+
+_Released <RELEASE_DATE> (PENDING)_
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
_Released 03/14/2023 (PENDING)_
```

**Review by emilyrohrbough — APPROVED** [review]

**Comment by emilyrohrbough** [comment]
@AlexGuironnetRTE thank you!!

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,13 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
```

**Review by mike-plummer — APPROVED** [review]

**Comment by mike-plummer** [comment]
@AlexGuironnetRTE Thank you for the contribution! 🎉 This will go out in our next release in the next two weeks or so

**Comment by AlexGuironnetRTE** [comment]
@emilyrohrbough @mike-plummer Awesome! Thanks a lot for integrating it so quickly!

**Pull request merged**